src/main/java/io/supertokens/webserver/api/session/SessionAPI.java

```diff
@@ -70,14 +70,24 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws I
         assert enableAntiCsrf != null;
         JsonObject userDataInJWT = InputParser.parseJsonObjectOrThrowError(input, "userDataInJWT", false);
         assert userDataInJWT != null;
+
         JsonObject userDataInDatabase = InputParser.parseJsonObjectOrThrowError(input, "userDataInDatabase", false);
         assert userDataInDatabase != null;
 
+        JsonObject grantPayload = super.getVersionFromRequest(req).equals("2.13")
+                ? InputParser.parseJsonObjectOrThrowError(input, "grants", false)
+                : null;
```

Comment:
I think here we should make it an empty JSON?

Reply:
That'd create a V3 token even with lower CDI versions. Which could break older backend SDKs.